### PR TITLE
Add OTEL_GO_AUTO_SHOW_VERIFIER_LOG env var

### PR DESCRIPTION
**Pull request by edeNFed**
Recent versions of cilium/ebpf library stopped printing detailed verifier logs by default.
This makes it hard to debug verifier errors (like #78).
This PR adds an additional env var called `OTEL_GO_AUTO_SHOW_VERIFIER_LOG` which will print the full verifier log if set to true.
Collecting verifier logs is an expensive operation, and therefore is turned off by default.

### CODE REVIEW

File: pkg/instrumentors/utils/ebpf.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	return err
+}
+
+// Getting full verifier log is expensive, so we only do it if the user explicitly asks for it.
+func shouldShowVerifierLogs() bool {
+	val, exists := os.LookupEnv(showVerifierLogEnvVar)
+	if exists {
+		boolVal, err := strconv.ParseBool(val)
+		if err != nil {
+			return true
```
Comment (MikeGoldsmith) [inline pkg/instrumentors/utils/ebpf.go:56]: Maybe we should default to `false` (disabled) if the value in the env var is invalid?

File: pkg/instrumentors/utils/ebpf.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const (
+	showVerifierLogEnvVar = "OTEL_GO_AUTO_SHOW_VERIFIER_LOG"
+)
+
+// LoadEbpfObjects loads eBPF objects from the given spec into the given interface.
+// If the environment variable OTEL_GO_AUTO_SHOW_VERIFIER_LOG is set to true, the verifier log will be printed.
+func LoadEbpfObjects(spec *ebpf.CollectionSpec, to interface{}, opts *ebpf.CollectionOptions) error {
+	showVerifierLogs := shouldShowVerifierLogs()
+	if showVerifierLogs {
+		opts.Programs.LogSize = ebpf.DefaultVerifierLogSize * 100
```
Comment (MikeGoldsmith) [inline pkg/instrumentors/utils/ebpf.go:35]: Why do we need to extend the default size? Should we consider allowing the log size to be adjusted too?

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
Look good to me. I've left a couple of queries regarding default values.

File: pkg/instrumentors/utils/ebpf.go (hunk 2 above)
Comment (MikeGoldsmith) [reply pkg/instrumentors/utils/ebpf.go:35]: Could we add a comment to share what you've put above, maybe as a declare var?

**Review by MikeGoldsmith — APPROVED** [review]
Looks good - I think a comment / issue to review LogSize configuration would be good 👍🏻

File: pkg/instrumentors/utils/ebpf.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/cilium/ebpf"
+)
+
+const (
+	showVerifierLogEnvVar = "OTEL_GO_AUTO_SHOW_VERIFIER_LOG"
+)
+
+// LoadEbpfObjects loads eBPF objects from the given spec into the given interface.
+// If the environment variable OTEL_GO_AUTO_SHOW_VERIFIER_LOG is set to true, the verifier log will be printed.
+func LoadEbpfObjects(spec *ebpf.CollectionSpec, to interface{}, opts *ebpf.CollectionOptions) error {
```
Comment (MrAlias) [inline pkg/instrumentors/utils/ebpf.go:32]: Hmm, I thought we had a lint check to catch this initialism. I'll have to follow up on that.

```suggestion
// LoadEBPFObjects loads eBPF objects from the given spec into the given interface.
// If the environment variable OTEL_GO_AUTO_SHOW_VERIFIER_LOG is set to true, the verifier log will be printed.
func LoadEBPFObjects(spec *ebpf.CollectionSpec, to interface{}, opts *ebpf.CollectionOptions) error {
```

File: pkg/instrumentors/utils/ebpf.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	showVerifierLogs := shouldShowVerifierLogs()
```
Comment (MrAlias) [inline pkg/instrumentors/utils/ebpf.go:33]:
```suggestion
	// Getting full verifier log is expensive, so we only do it if the user explicitly asks for it.
	showVerifierLogs := shouldShowVerifierLogs()
```

File: pkg/instrumentors/utils/ebpf.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		var ve *ebpf.VerifierError
+		if errors.As(err, &ve) {
+			fmt.Printf("Verifier log: %+v\n", ve)
+		}
+	}
+
+	return err
+}
+
+// Getting full verifier log is expensive, so we only do it if the user explicitly asks for it.
```
Comment (MrAlias) [inline pkg/instrumentors/utils/ebpf.go:50]:
```suggestion
// shouldShowVerifierLogs returns if the user has configured verifier logs to be emitted.
```

**Pull request merged**